float64 scale
---
bool success
